src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -41,6 +42,34 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
                     throw new NuGetResolverInputException(String.Format(CultureInfo.CurrentCulture, Strings.MissingDependencyInfo, requiredId));
                 }
             }
+            
+            var invalidsPackages = new List<string>();
+
+            var installedPackages = context.PackagesConfig.Select(p => p.PackageIdentity).ToArray();
+
+            // validate existing package.config for any invalid dependency
+            foreach (var package in installedPackages)
+            {
+                var existingPackage = context.AvailablePackages.FirstOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.Id, package.Id) && p.Version.Equals(package.Version));
+
+                if (existingPackage != null)
```

Comment:
As we talked about, this shouldn't happen. it's up to you if you want to add the message.